polkadot/node/network/collator-protocol/src/validator_side_experimental/common.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use polkadot_node_network_protocol::peer_set::CollationVersion;
+use polkadot_primitives::Id as ParaId;
+
+/// Maximum reputation score.
+pub const MAX_SCORE: u16 = 1000;
+
+/// Limit for the total number connected peers.
+pub const CONNECTED_PEERS_LIMIT: u16 = 300;
+
+/// Limit for the total number of connected peers for a paraid.
+/// Must be smaller than `CONNECTED_PEERS_LIMIT`.
+pub const CONNECTED_PEERS_PARA_LIMIT: u16 = 100;
```

Comment:
This is quite a lot, given that we disconnect them on inactivity and they connect only when they have a collation.

Reply:
we don't disconnect on inactivity. While collators indeed currently only connect when they have a collation, we could have them connect in advance to lower the backing latency.
Disconnecting on inactivity complicates things and I don't see a clear benefit

Reply:
Even if we have them connecting in advance up to claim queue size blocks, this is still a large value. I would try to minimize the amount of resources, so having it would be good to have some sort of punishment for connecting and chilling.